serde_v8/src/de.rs

```diff
@@ -338,57 +339,19 @@ impl<'de, 'a, 'b, 's, 'x> de::Deserializer<'de>
   where
     V: Visitor<'de>,
   {
-    // Magic for serde_v8::magic::Value, to passthrough v8::Value
-    // TODO: ensure this is cross-platform and there's no alternative
-    if name == magic::NAME {
-      let mv = magic::Value {
-        v8_value: self.input,
-      };
-      let hack: u64 = unsafe { std::mem::transmute(mv) };
-      return visitor.visit_u64(hack);
-    }
-
-    // Magic Buffer
-    if name == magic::buffer::BUF_NAME {
-      let zero_copy_buf = match self.input.is_array_buffer() {
-        // ArrayBuffer
-        true => v8::Local::<v8::ArrayBuffer>::try_from(self.input)
-          .and_then(magic::zero_copy_buf::ZeroCopyBuf::try_from),
-        // maybe ArrayBufferView
-        false => v8::Local::<v8::ArrayBufferView>::try_from(self.input)
-          .and_then(|view| {
-            magic::zero_copy_buf::ZeroCopyBuf::try_from((
-              &mut *self.scope,
-              view,
-            ))
-          }),
-      }
-      .map_err(|_| Error::ExpectedBuffer)?;
-      let data: [u8; 32] = unsafe { std::mem::transmute(zero_copy_buf) };
-      return visitor.visit_bytes(&data);
-    }
-
-    // Magic ByteString
-    if name == magic::bytestring::NAME {
-      let v8str = v8::Local::<v8::String>::try_from(self.input)
-        .map_err(|_| Error::ExpectedString)?;
-      if !v8str.contains_only_onebyte() {
-        return Err(Error::ExpectedLatin1);
-      }
-      let len = v8str.length();
-      let mut buffer = Vec::with_capacity(len);
-      #[allow(clippy::uninit_vec)]
-      unsafe {
-        buffer.set_len(len);
-      }
-      let written = v8str.write_one_byte(
-        self.scope,
-        &mut buffer,
-        0,
-        v8::WriteOptions::NO_NULL_TERMINATION,
-      );
-      assert!(written == len);
-      return visitor.visit_byte_buf(buffer);
+    // Magic types
+    if name == Buffer::magic_name() {
```

Comment:
Maybe match + implicit return expressions?

Reply:
Match arms can't have function calls and these values can't yet be associated attributes because `std::any::type_name::<T>()` isn't yet const, see https://github.com/rust-lang/rust/issues/63084

Obviously could do this semi-manually but it's not a huge deal and relatively unimportant